content/articles/image-gallery-flask-cloudinary-mongo/index.md

```diff
@@ -0,0 +1,385 @@
+### Building a CDN Image Gallery with Flask, Cloudinary, and MongoDB
+
+### Introduction
+
+A content delivery network (CDN) allows for the quick transfer and retrieval of web assets by distributing them to geographically distant servers that work together to provide required content to end-users.
+
+Content delivery networks (CDNs) work by hosting files (images, videos, HTML pages, stylesheets, JavaScript files, etc.) in many servers spread across the globe. When a user requests the file, the CDN loads it from the server closest to the user location (among other optimizations), reducing latency in fetching the assets and improving content availability.
+
+In this article, you will learn:
+
+* Benefits of using content delivery networks (CDNs).
+* How to build CRUD applications with Flask and MongoDB.
+* How to integrate Cloudinary services into a Python application.
+
+### Benefits of Content Delivery Networks (CDNs)
+
+#### Reduction in Website Loading Time
+
+Since CDNs fetch their assets from a server close to the user (among many other optimization techniques), making websites faster for visitors.
+
+#### Improved Content Availability and Redundancy
+
+CDNs share their files to multiple servers simultaneously, so you are sure server files are available at any time, even if there’s a server outage. CDNs can also manage large amounts of traffic and withstand hardware failure better than the origin server.
+
+#### Improved Website Security
+
+CDNs improve websites’ safety by providing [DDoS attack](https://www.section.io/engineering-education/denial-of-service/) mitigation, improving server security certificates, and other optimizations.
+
+#### Reduction of Server Bandwidth
+
+CDN assets are on multiple servers. This reduces the amount of data the origin server has to load and cache. It also lowers hosting costs incurred from bandwidth consumption.
+
+### Building the Image Gallery with Flask and MongoDB
+
+In this tutorial, you will build an image gallery with Flask and MongoDB that integrates a CDN with [Cloudinary](https://cloudinary.com/documentation/cloudinary_get_started). Create the backend with [Flask](https://flask.palletsprojects.com/en/1.1.x/), database with [Flask-PyMongo](https://flask-pymongo.readthedocs.io/en/latest/), and user interface with [Flask-Bootstrap](https://pypi.org/project/Flask-Bootstrap4/).
+
+You will build three (3) web pages for the image gallery. They are:
+
+* Index/Landing Page
+* Upload Image Page
+* View Gallery Page
+
+#### Step 1: Installing the App Requirements
+
+Install the libraries required to build the image gallery like Flask, PyMongo, and Flask-Bootstrap. In your terminal, type:
```

Comment:
You can add another block explaining briefly what these packages do.

Reply:
okay thanks, I'll make appropriate changes